File: 7-reverse-integer/7-reverse-integer.py
class Solution:
    def reverse(self, x: int) -> int:
        if x < 0:
            x = abs(x)
            s = str(x)
            x = -1 * int(s[-1::-1])
        else:
            s = str(x)
            x = int(s[-1::-1])
        return x if (x >= -2**31) and (x <= (2**31)-1) else 0
        

File: 7-reverse-integer/7-reverse-integer.cpp
class Solution {
public:
    int reverse(int x) {
        long long int num = x;
        long long int ans = 0;
        
        while(num){
            ans = ans*10 + num%10;
            num = num/10;
        }
        
        if(ans < INT_MIN || ans > INT_MAX) return 0;
        return ans;
    }
};